napari/components/add_layers_mixin.py

```diff
@@ -849,41 +850,29 @@ def _add_layers_with_plugins(
         """
         layer_data = read_data_with_plugins(path_or_paths, plugin=plugin)
 
-        if not layer_data:
-            # if layer_data is empty, it means no plugin could read path
-            # we just want to provide some useful feedback, which includes
-            # whether or not paths were passed to plugins as a list.
-            if isinstance(path_or_paths, (tuple, list)):
-                path_repr = f"[{path_or_paths[0]}, ...] as stack"
+        # glean layer names from filename. These will be used as *fallback*
+        # names, if the plugin does not return a name kwarg in their meta dict.
+        if isinstance(path_or_paths, str):
+            filenames = itertools.repeat(path_or_paths)
+        elif is_sequence(path_or_paths):
+            if len(path_or_paths) == len(layer_data):
+                filenames = iter(path_or_paths)
             else:
-                path_repr = path_or_paths
-            msg = f'No plugin found capable of reading {path_repr}.'
-            logger.error(msg)
-            return []
+                # if a list of paths has been returned as a list of layer data
+                # without a 1:1 relationship between the two lists we iterate
+                # over the first name
+                filenames = itertools.repeat(path_or_paths[0])
```

Comment:
Wow this wouldn’t have occurred to me to check for... do we really not have a way to map these correctly?

Reply:
not really, there's nothing (currently) stopping a plugin from returning a different number of layer_data tuples than there were paths in the list.  probably unlikely, but possible.  in most cases, I suspect it will be a listed of files stacked into a single image...

Reply:
Well, I don't see a way around this, but it's making me deeply uncomfortable, for the record. =P